io/src/main/scala/fs2/io/udp/AsynchronousSocketGroup.scala

```diff
@@ -32,55 +35,93 @@ object AsynchronousSocketGroup {
 
   def apply(): AsynchronousSocketGroup = new AsynchronousSocketGroup {
 
+    class Timeout(val expiry: Long, onTimeout: () => Unit) {
+      private var done: Boolean = false
```

Comment:
Yep, it is only read and written by the single UDP selector thread. Never from any other thread.